openshift_cli_installer/libs/clusters/aro_cluster.py

```diff
@@ -0,0 +1,248 @@
+from openshift_cli_installer.libs.clusters.ocp_cluster import OCPCluster
+from simple_logger.logger import get_logger
+
+
+class AROCluster(OCPCluster):
+    def __init__(self, ocp_cluster, user_input):
+        super().__init__(ocp_cluster=ocp_cluster, user_input=user_input)
+        self.logger = get_logger(f"{self.__class__.__module__}-{self.__class__.__name__}")
+        self.set_azure_authentication()
+
+        if not self.user_input.create:
+            with open(self._cluster_data_yaml_file, "r") as fd:
+                cluster_data = yaml.safe_load(stream=fd)
+                self.set_aro_cluster_params(cluster_data=cluster_data["cluster"])
+        else:
+            self.set_aro_cluster_params(cluster_data=self.cluster_info)
+
+    def set_azure_authentication(self):
+        self.tenant_id = self.user_input.azure_tenant_id
+        self.client_id = self.user_input.azure_client_id
+        self.client_secret = self.user_input.azure_client_secret
+        self.subscription_id = self.user_input.azure_subscription_id
+        self.credentials = azure_credentials(
+            tenant_id=self.tenant_id,
+            client_id=self.client_id,
+            client_secret=self.client_secret,
+        )
+
+        self.aro_client, self.network_client, self.resource_client = (
+            client(credential=self.credentials, subscription_id=self.subscription_id)
+            for client in (aro_client, network_client, resource_client)
+        )
+
+    def set_aro_cluster_params(self, cluster_data):
+        self.cluster_name = self.get_cluster_name()
+        cluster_resources_postfix = random_resource_postfix()
+        resource_group_name_str = "resource-group-name"
+        virtual_network_name_str = "virtual-network-name"
+
+        aro_cluster_args = {
+            "domain": "msi",
+            resource_group_name_str: f"aro-rg-{cluster_resources_postfix}",
+            "cluster-resource-group-name": f"{self.cluster_name}-rg-{cluster_resources_postfix}",
+            virtual_network_name_str: f"aro-vnet-{cluster_resources_postfix}",
+            "workers-subnet-name": f"workers-subnet-{cluster_resources_postfix}",
+            "master-subnet-name": f"master-subnet-{cluster_resources_postfix}",
+            "master-vm-size": "Standard_D8s_v3",
+            "workers-vm-size": "Standard_D4s_v3",
+            "workers-count": 3,
+            "workers-disk-size": 128,
+            "network-pod-cidr": "10.128.0.0/14",
+            "network-service-cidr": "172.30.0.0/16",
+            "fips": False,
+        }
+        self.cluster.update({k: cluster_data.pop(k, v) for k, v in aro_cluster_args.items()})
+
+        if pull_secret_file := self.user_input.docker_config_file:
+            with open(pull_secret_file, "r") as ps_file:
+                self.pull_secret = ps_file.read()
+
+        self.region = self.cluster["region"]
+        self.version = self.cluster["version"]
+        self.platform = self.cluster_info["platform"]
```

Comment:
consume directly where needed, no need to assign

Reply:
done